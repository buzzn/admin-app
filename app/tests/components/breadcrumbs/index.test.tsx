import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Breadcrumbs from 'components/breadcrumbs';

describe('breadcrumbs tests', () => {
  test('renders different crumb types', () => {
    const breadcrumbs = [
      { id: 0, title: 'Title' },
      { id: 1, type: 'meter', title: 'Title' },
      { id: 2, link: 'localhost', title: 'Title' },
      { id: 2, type: 'meter', link: 'localhost', title: 'Title' },
    ];
    const tree = renderer
      .create(<MemoryRouter>
          <Breadcrumbs {...{ breadcrumbs }} />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
