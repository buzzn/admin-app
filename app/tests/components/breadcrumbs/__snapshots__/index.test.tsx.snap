// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs tests renders different crumb types 1`] = `
<div
  className="sc-bdVaJa RchYe"
>
  <ul>
    <li>
      <span>
        Title
      </span>
    </li>
    <li>
      <span>
        <span>
          <span
            className="breadcrumb-type"
          >
            admin.types.meter
            :
          </span>
           
          Title
        </span>
      </span>
    </li>
    <li>
      <span>
        <a
          href="/localhost"
          onClick={[Function]}
        >
          Title
        </a>
         / 
      </span>
    </li>
    <li>
      <span>
        <a
          href="/localhost"
          onClick={[Function]}
        >
          <span>
            <span
              className="breadcrumb-type"
            >
              admin.types.meter
              :
            </span>
             
            Title
          </span>
        </a>
         / 
      </span>
    </li>
  </ul>
</div>
`;
